Content.Server/GameObjects/EntitySystems/Weapon/Melee/MeleeWeaponSystem.cs

```diff
@@ -0,0 +1,267 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Content.Server.GameObjects.Components.Body.Circulatory;
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.GameObjects.Components.Weapon.Melee;
+using Content.Shared.GameObjects.Components.Damage;
+using Content.Shared.GameObjects.Components.Items;
+using Content.Shared.GameObjects.EntitySystemMessages;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Content.Shared.Physics;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Broadphase;
+using Robust.Shared.Player;
+using Robust.Shared.Timing;
+
+namespace Content.Server.GameObjects.EntitySystems.Weapon.Melee
+{
+    public sealed class MeleeWeaponSystem : EntitySystem
+    {
+        [Dependency] private IGameTiming _gameTiming = default!;
+        [Dependency] private IEntityManager _entityManager = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<MeleeWeaponComponent, HandSelectedEvent>(OnHandSelected);
+            SubscribeLocalEvent<MeleeWeaponComponent, ClickAttackEvent>(OnClickAttack);
+            SubscribeLocalEvent<MeleeWeaponComponent, WideAttackEvent>(OnWideAttack);
+            SubscribeLocalEvent<ItemCooldownComponent, RefreshItemCooldownEvent>(OnCooldownRefreshed);
+
+            SubscribeLocalEvent<MeleeChemicalInjectorComponent, MeleeHitEvent>(OnChemicalInjectorHit);
+        }
+
+        private void OnHandSelected(EntityUid uid, MeleeWeaponComponent comp, HandSelectedEvent args)
+        {
+            var curTime = _gameTiming.CurTime;
+            var cool = TimeSpan.FromSeconds(comp.CooldownTime * 0.5f);
+
+            if (curTime < comp.CooldownEnd)
+            {
+                if (comp.CooldownEnd - curTime < cool)
+                {
+                    comp.LastAttackTime = curTime;
+                    comp.CooldownEnd += cool;
+                }
+                else
+                    return;
+            }
+            else
+            {
+                comp.LastAttackTime = curTime;
+                comp.CooldownEnd = curTime + cool;
+            }
+
+            RaiseLocalEvent(uid, new RefreshItemCooldownEvent(comp.LastAttackTime, comp.CooldownEnd));
+        }
+
+        private void OnClickAttack(EntityUid uid, MeleeWeaponComponent comp, ClickAttackEvent args)
+        {
+            var curTime = _gameTiming.CurTime;
+            if (!_entityManager.TryGetEntity(uid, out var owner))
+            {
+                args.Succeeded = false;
+                return;
+            }
```

Comment:
Don't `TryGetEntity` here, you can just use `GetEntity` as the entity ALWAYS exists in directed event callbacks

Reply:
Also EntityManager is already a property on systems.